service/storage_service.cc

```diff
@@ -2341,6 +2342,11 @@ future<> storage_service::removenode(sstring host_id_string, std::list<gms::inet
             auto endpoint = *endpoint_opt;
             auto tokens = tmptr->get_tokens(endpoint);
             auto leaving_nodes = std::list<gms::inet_address>{endpoint};
+            std::list<gms::inet_address> ignore_nodes;
```

Comment:
If we parse the host_id in api/, we can keep the strong type for ignore_nodes parameter.

Reply:
what about the host_id? it's also passed as a string so we should do all the parsing in the api layer or here.
I prefer to do it here since we have the token_metadata at hand, while the api layer doesn't know about it and it's current version.

Reply:
We can convert the host_id to uuid and pass too. But  see my comments above, we can do the parse in token_metada.